framework/vue/router.ts

```diff
@@ -0,0 +1,174 @@
+import { Component, createApp, createSSRApp, defineComponent, h } from "vue";
+import type { SSRContext } from "../../server/renderer.ts";
+import { DataContext, RouterContext } from "./context.ts";
+import { RouteModule } from "../core/route.ts";
+import { Link } from "./link.ts";
+import { Head } from "./head.ts";
+
+// deno-lint-ignore no-explicit-any
+const global = window as any;
+
+type RouteData = {
+  data?: unknown;
+  dataCacheTtl?: number;
+  dataExpires?: number;
+};
+
+type RootProps = {
+  ssrContext?: SSRContext;
+};
+
+export const App = defineComponent({
+  name: "App",
+  props: {
+    ssrContext: {
+      type: String,
+      default: "",
+    },
+  },
+  setup() {
+    console.log("App setup");
+  },
+  mounted() {
+    console.log("App mounted");
+  },
+  render() {
+    return this.$slots.default ? this.$slots.default() : [];
+  },
+});
+
+export const Err = defineComponent({
+  name: "Err",
+  props: {
+    status: {
+      type: String,
+      default: "404",
+    },
+  },
+  render() {
+    return h("div", {
```

Comment:
Seems those code can be shared between react and vue, we will figure out

Reply:
> Seems those code can be shared between react and vue, we will figure out

I think we can make `share/types` dir for framework, it can help `react` and `vue`. it alse help other framework like `sevlet` in the feature.